infra/upload_images/container_ocr/handler/embedding_processor.py

```
@@ -304,62 +317,64 @@ def _resolve_merchant(
-                openai_client = get_client_manager().openai
-                resp = openai_client.embeddings.create(
-                    model="text-embedding-3-small",
-                    input=list(texts)
+
+            # Create embedding function for ChromaDB (if available)
+            if chroma_line_client:
+                def _embed_texts(texts):
+                    if not texts:
+                        return []
+                    if not self.openai_api_key:
+                        raise RuntimeError("OPENAI_API_KEY is not set")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)